File: Counting-Sort-2.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'countingSort' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static List<int> countingSort(List<int> arr)
    {
        if (arr.Count == 0) return arr;

        int max = arr.Max();
        int min = arr.Min();
        int range = max - min + 1;

        int[] count = new int[range];
        int[] output = new int[arr.Count];

        foreach (int num in arr)
            count[num - min]++;

        for (int i = 1; i < count.Length; i++)
            count[i] += count[i - 1];

        for (int i = arr.Count - 1; i >= 0; i--)
        {
            int num = arr[i];
            output[count[num - min] - 1] = num;
            count[num - min]--;
        }

        return new List<int>(output);
    }
}


class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> arr = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arrTemp => Convert.ToInt32(arrTemp)).ToList();

        List<int> result = Result.countingSort(arr);

        textWriter.WriteLine(String.Join(" ", result));

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Counting-Sort-2.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'countingSort' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static List<Integer> countingSort(List<Integer> arr) {
        if (arr.isEmpty()) return arr;

        int max = Collections.max(arr);
        int min = Collections.min(arr);
        int range = max - min + 1;

        int[] count = new int[range];
        int[] output = new int[arr.size()];

        for (int num : arr)
            count[num - min]++;

        for (int i = 1; i < count.length; i++)
            count[i] += count[i - 1];

        for (int i = arr.size() - 1; i >= 0; i--) {
            int num = arr.get(i);
            output[count[num - min] - 1] = num;
            count[num - min]--;
        }

        List<Integer> sorted = new ArrayList<>();
        for (int num : output)
            sorted.add(num);

        return sorted;
    }
}
